pkg/controller/controller_test.go

```diff
@@ -1379,3 +1322,94 @@ func assertEmptyFinalizers(t *testing.T, obj runtime.Object) {
 		fatalf(t, "Unexpected number of finalizers; expected 0, got %v", len(accessor.GetFinalizers()))
 	}
 }
+
+func assertNumberOfBrokerActions(t *testing.T, actions []fakeosb.Action, number int) {
+	testNumberOfBrokerActions(t, "" /* name */, fatalf, actions, number)
+}
+
+func expectNumberOfBrokerActions(t *testing.T, name string, actions []fakeosb.Action, number int) bool {
+	return testNumberOfBrokerActions(t, name, errorf, actions, number)
+}
+
+func testNumberOfBrokerActions(t *testing.T, name string, f failfFunc, actions []fakeosb.Action, number int) bool {
+	logContext := ""
+	if len(name) > 0 {
+		logContext = name + ": "
+	}
+
+	if e, a := number, len(actions); e != a {
+		t.Logf("%+v\n", actions)
+		f(t, "%vUnexpected number of actions: expected %v, got %v", logContext, e, a)
+		return false
+	}
+
+	return true
+}
+
+func noFakeActions() fakeosb.FakeClientConfiguration {
+	return fakeosb.FakeClientConfiguration{}
+}
+
+func assertGetCatalog(t *testing.T, action fakeosb.Action) {
+	if e, a := fakeosb.GetCatalog, action.Type; e != a {
+		fatalf(t, "unexpected action type; expected %v, got %v", e, a)
+	}
+}
+
+func assertProvision(t *testing.T, action fakeosb.Action, request *osb.ProvisionRequest) {
+	if e, a := fakeosb.ProvisionInstance, action.Type; e != a {
+		fatalf(t, "unexpected action type; expected %v, got %v", e, a)
+	}
+
+	if e, a := request, action.Request; !reflect.DeepEqual(e, a) {
+		fatalf(t, "unexpected diff in provision request: %v\nexpected %+v\ngot      %+v", diff.ObjectReflectDiff(e, a), e, a)
```

Comment:
nit: can you remove all those spaces in front of the 2nd `%+v`? maybe replace them with a newline